tests/epyccel/test_default_precision_template.py

```diff
@@ -0,0 +1,29 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring
+
+from numpy.random import randint
+from numpy import isclose
+import numpy as np
+
+from pyccel.decorators import types
+from pyccel.epyccel import epyccel
+
+
+def test_default_precision_template(language):
+
+    @types('int[:]')
+    @types('float[:]')
+    @types('complex[:]')
+    def return_array_element(array):
+        return array[0]
+
+    test_types = ['int', 'float', 'complex']
+    f1 = return_array_element
+    f2 = epyccel(f1, language=language)
+    for t in test_types:
+        d1 = randint(1, 15)
+        arr = np.ones(d1, dtype=t)
+        python_result = f1(arr)
+        pyccel_result = f2(arr)
+
+        assert isinstance(pyccel_result, type(python_result))
+        assert isclose(pyccel_result, python_result)
```

Comment:
In the unit tests we should explicitly define the relative and absolute tolerances for Numpy array comparison, by passing `rtol` and `atol`. See [here](https://github.com/pyccel/pyccel/blob/ca37095162bf364b63abf88f956d66e5bf9bb248/tests/epyccel/test_arrays.py#L3970-L4012) for example.